### PR TITLE
LSR-885 Change all cache operations to background

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

File: ios/Framework/Core/Offline/CacheManager.swift [hunk 1]
```diff
@@ -54,31 +54,42 @@ public enum CacheStrategyType: String {
...
 	public func getString(collection collection: String, key: String, result: String? -> ()) {
-		readConnection.readWithBlock { transaction in
-			let value: AnyObject? = transaction.objectForKey(key, inCollection: collection)
-			result((value as? NSObject)?.description)
-		}
+
+		var value: AnyObject?
+		readConnection.asyncReadWithBlock( { transaction in
+				value = transaction.objectForKey(key, inCollection: collection)
+			}) {
```
Comment (jmnavarro) [inline ios/Framework/Core/Offline/CacheManager.swift]: when you have more than one closure, don't use the trailing closure syntax. It's quite confusing...

File: ios/Framework/Core/Auth/UserPortraitScreenlet/Interactors/UploadUserPortraitInteractor.swift [hunk 2]
```diff
@@ -54,7 +54,8 @@ class UploadUserPortraitInteractor: ServerWriteConnectorInteractor {
 			collection: ScreenletName(UserPortraitScreenlet),
 			key: "userId-\(userId)",
 			value: image,
-			attributes: ["userId": userId.description])
+			attributes: ["userId": userId.description],
+			onCompletion: nil)
```
Comment (jmnavarro) [inline ios/Framework/Core/Auth/UserPortraitScreenlet/Interactors/UploadUserPortraitInteractor.swift]: if there's a default value, don't include this

Comment (victorg1991) [inline ios/Framework/Core/Auth/UserPortraitScreenlet/Interactors/UploadUserPortraitInteractor.swift]: If you omit this in this case there is a compiler error

**Comment by jmnavarro** [comment]
👍

**Pull request merged**